komodo-relational/src/main/java/org/komodo/relational/datasource/internal/DatasourceParser.java

```diff
@@ -0,0 +1,445 @@
+                
+                Datasource theDatasource = null;
+                if(existingSource!=null && existingSource.length>0) {
+                    if(replaceExisting) {
+                        if(deleteDatasource(existingSource[0])) {
+                            theDatasource = createDatasource();
+                            if(theDatasource!=null) dataSources.add(theDatasource);
+                        }
+                    }
+                } else {
+                    theDatasource = createDatasource();
+                    if(theDatasource!=null) dataSources.add(theDatasource);
+                }
+                                
+                this.datasourceName = null;
+                this.jdbc = null;
+                this.propertyMap.clear();
+            } else if(Datasource.XML_ELEM_PROPERTY.equals(localName)) {
+                this.propertyMap.put(this.propertyName, this.propertyValue.toString());
+                this.propertyName = null;
+                this.propertyValue.setLength(0);
+            }
+
+            this.elements.pop();
+
+            super.endElement(uri, localName, qName);
+        }
+
+        private boolean deleteDatasource(KomodoObject source) {
+            try {
+                WorkspaceManager.getInstance( repo ).delete(uow, source);
+            } catch (KException ex) {
+                error(new SAXParseException(ex.getMessage(), null));
+                LOGGER.error("DatasourceParser - error deleting existing Datasource : ", ex); //$NON-NLS-1$
+                return false;
+            }
+            return true;
+        }
+        
+        private Datasource createDatasource() {
+            Datasource theDatasource = null;
+            try {
+                theDatasource = RelationalModelFactory.createDatasource(uow, repo, repo.komodoWorkspace( uow ).getAbsolutePath(), datasourceName);
+                
+                // Set jdbc from attribute
+                boolean isJdbc = true;
+                if( this.jdbc != null ) {
+                    isJdbc = Boolean.parseBoolean(this.jdbc);
+                }
+                theDatasource.setJdbc(uow, isJdbc);
+                
+                // Set remaining properties
+                setDatasourceProperties(uow, theDatasource, this.propertyMap);
+            } catch (KException ex) {
+                error(new SAXParseException(ex.getMessage(), null));
+                LOGGER.error("DatasourceParser - error creating Datasource : ", ex); //$NON-NLS-1$
+            }
+            return theDatasource;
+        }
+        
+        // Sets the Datasource properties using the provided map
+        private void setDatasourceProperties( final UnitOfWork transaction, final Datasource datasource, final Map<String,String> propMap ) throws KException {
+            int nsPrefixLength = (KomodoLexicon.Namespace.PREFIX+StringConstants.COLON).length();
+            
+            for(String propName : propMap.keySet()) {
```

Comment:
What do you think about overriding the `setProperty` method in `Datasource` and moving this logic there? So inside this loop you would just call that new method.